src/DefaultBuilder/src/ConfigureWebHostBuilder.cs

```diff
@@ -153,5 +154,20 @@ public IWebHostBuilder UseSetting(string key, string? value)
 
             return this;
         }
+
+        IWebHostBuilder ISupportsStartup.Configure(Action<WebHostBuilderContext, IApplicationBuilder> configure)
+        {
+            throw new NotSupportedException($"Configure() is not supported by WebApplicationBuilder.WebHost. Use the WebApplication returned by WebApplicationBuilder.Build() instead.");
+        }
+
+        IWebHostBuilder ISupportsStartup.UseStartup(Type startupType)
+        {
+            throw new NotSupportedException($"UseStartup() is not supported by WebApplicationBuilder.WebHost. Use the WebApplication returned by WebApplicationBuilder.Build() instead.");
```

Comment:
I think these messages probably need work. Someone could also be interested in modifying services with the startup class, and for that they want to use `WebApplicationBuilder.Services` instead of `WebApplication`'s `IApplicationBuilder` implementation. I didn't want this to get too wordy though.